libraries/AP_NavEKF2/AP_NavEKF2_Measurements.cpp

```diff
@@ -195,8 +195,9 @@ void NavEKF2_core::readMagData()
     }
     // If we are a vehicle with a sideslip constraint to aid yaw estimation and we have timed out on our last avialable
     // magnetometer, then declare the magnetometers as failed for this flight
-    uint8_t maxCount = _ahrs->get_compass()->get_count();
-    if (allMagSensorsFailed || (magTimeout && assume_zero_sideslip() && magSelectIndex >= maxCount-1 && inFlight)) {
+    if (allMagSensorsFailed || (magTimeout && assume_zero_sideslip() && 
+                                (magNumTimesChanged >= (_ahrs->get_compass()->num_detected() - 1)) && 
```

Comment:
I don't understand the equivalence of this code. magNumTimesChanged is always increasing, and could quickly get beyond num_detected. I think this will kill the EKF after a few compass changes